lib/Sema/TypeCheckAttr.cpp

```diff
@@ -7133,6 +7135,25 @@ void AttributeChecker::visitMacroRoleAttr(MacroRoleAttr *attr) {
     break;
   }
   }
+
+  for (auto *typeExpr : attr->getConformances()) {
+    if (auto *typeRepr = typeExpr->getTypeRepr()) {
+      auto *dc = D->getDeclContext();
+      auto resolved =
+          TypeResolution::forInterface(
+              dc, TypeResolverContext::GenericRequirement,
+              /*unboundTyOpener*/ nullptr,
+              /*placeholderHandler*/ nullptr,
+              /*packElementOpener*/ nullptr)
+          .resolveType(typeRepr);
+
+      if (resolved->is<ErrorType>()) {
+        attr->setInvalid();
+      } else {
+        typeExpr->setType(MetatypeType::get(resolved));
```

Comment:
We'll need to check that this refers to a protocol type (or, I suppose, a protocol composition type).